### PR TITLE
Feature/allocation score

### PR DESCRIPTION
**Pull request by BenDundon**
Adds a data type to score an allocation of an academic to a unit. Written this time in TypeScript as that's what Adonis expects according to their docs, however won't be fully testable until CPF-36 completes as nothing calls this code yet.

### CODE REVIEW

File: tas/app/Models/formResponses.ts
Comment (TsubasaBneAus) [inline tas/app/Models/formResponses.ts:1]: @BenDundon 
- In the comments, the values of the scales for **willingness** and **expertise** should be changed to the 1-10 scale as I mentioned in **academic preference form** branch.
- Indent size should be changed to 2 using like **Prettier** to keep all files have the same indent size

File: tas/app/Controllers/Http/AutoAllocatorController.ts
Comment (TsubasaBneAus) [inline tas/app/Controllers/Http/AutoAllocatorController.ts:1]: @BenDundon 
- You should format this script to organise the codes using **Prettier**, and the indent size should be changed to 2 that is the same as its size in other source codes.

**Review by TsubasaBneAus — CHANGES_REQUESTED** [review]
@BenDundon 
I have reviewed your codes and left some comments on them. Please have a look at them!

File: tas/.env.example
Comment (TsubasaBneAus) [inline tas/.env.example:1]: @BenDundon 
The value of **HOST** in the .env.example file should be changed to **localhost** to make it easier for developers to understand how the server of this app runs.
This issue is not related to the main problem, but it is better to change it.

**Review by TsubasaBneAus — CHANGES_REQUESTED** [review]

File: tas/.env.example
Comment (BenDundon) [reply tas/.env.example:1]: Hi @TsubasaBneAus, `0.0.0.0` refers the same as `localhost`, as it directs to the **default route**, which on a user device is `127.0.0.1` (`localhost`).

**Pull request merged**